File: jar-persistence/src/test/java/com/sixsq/slipstream/event/EventTest.java
package com.sixsq.slipstream.event;

import java.util.Arrays;
import java.util.Date;
import java.util.List;

import com.sixsq.slipstream.acl.ACL;
import com.sixsq.slipstream.acl.TypePrincipal;
import com.sixsq.slipstream.acl.TypePrincipalRight;
import org.junit.Assert;
import org.junit.Test;

import com.sixsq.slipstream.event.Event.EventType;
import com.sixsq.slipstream.acl.TypePrincipal.PrincipalType;
import com.sixsq.slipstream.acl.TypePrincipalRight.Right;


public class EventTest {

	@Test
	public void eventCanBeCreatedAndJsonified() {
		
		TypePrincipal owner = new TypePrincipal(PrincipalType.USER, "joe");
		List<TypePrincipalRight> rules = Arrays.asList(new TypePrincipalRight(
				PrincipalType.ROLE, "ANON", Right.ALL));
		ACL acl = new ACL(owner, rules);

		Event event = new Event(acl, new Date(), "run/12313", "Initializing", Event.Severity.medium,
				EventType.state);
		
		Assert.assertNotNull(event.toJson());
	}

}

File: ssclj/jar/test/com/sixsq/slipstream/ssclj/resources/spec/event_test.clj
(ns com.sixsq.slipstream.ssclj.resources.spec.event-test
  (:require
    [clojure.test :refer [deftest are is]]
    [clojure.spec.alpha :as s]
    [com.sixsq.slipstream.ssclj.resources.event :refer :all]
    [com.sixsq.slipstream.ssclj.resources.common.crud :as crud]))

(def event-timestamp "2015-01-16T08:05:00.0Z")

(def valid-event {
                  :acl         {
                                :owner {
                                        :type "USER" :principal "joe"}
                                :rules [{:type "ROLE" :principal "ANON" :right "ALL"}]}

                  :id          "event/262626262626262"

                  :resourceURI resource-uri

                  :timestamp   event-timestamp
                  :content     {
                                :resource {:href "module/HNSciCloud-RHEA/S3"}
                                :state    "Started"}
                  :type        "state"
                  :severity    "critical"
                  })

(defn valid? [event] (s/valid? :cimi/event event))
(def invalid? (complement valid?))

(deftest check-event
  (is (= valid-event (crud/validate valid-event)))
  (is (valid? valid-event)))

(deftest check-reference
  (let [updated-event (assoc-in valid-event [:content :resource :href] "another/valid-identifier")]
    (is (= updated-event (crud/validate updated-event)))
    (is (valid? updated-event)))
  (let [updated-event (assoc-in valid-event [:content :resource :href] "/not a valid reference/")]
    (is (invalid? updated-event))))

(deftest check-severity
  (doseq [valid-severity ["critical" "high" "medium" "low"]]
    (is (valid? (assoc valid-event :severity valid-severity))))
  (is (invalid? (assoc valid-event :severity "unknown-severity"))))

(deftest check-type
  (doseq [valid-type ["state" "alarm"]]
    (is (valid? (assoc valid-event :type valid-type))))
  (is (invalid? (assoc valid-event :type "unknown-type"))))
